qubes/tests/ext.py

```diff
@@ -322,3 +330,38 @@ def test_012_supported_services_remove(self):
         self.assertEqual(self.features, {
             'supported-service.test2': True,
         })
+
+    def test_013_feature_set_dom0(self):
+        self.test_base_dir = '/tmp/qubes-test-dir'
+        self.base_dir_patch = mock.patch.dict(
+            qubes.config.system_path, {'dom0_services_dir': self.test_base_dir})
+        self.base_dir_patch.start()
+        service = 'guivm-gui-agent'
+        service_path = self.test_base_dir + '/' + service
+
+        self.ext.on_domain_feature_set(
+            self.dom0,
+            'feature-set:service.service.guivm-gui-agent',
+            'service.guivm-gui-agent', '1')
+
+        self.assertEqual(os.path.exists(service_path), True)
+
+    def test_014_feature_delete_dom0(self):
+        self.test_base_dir = '/tmp/qubes-test-dir'
+        self.base_dir_patch = mock.patch.dict(
+            qubes.config.system_path, {'dom0_services_dir': self.test_base_dir})
+        self.base_dir_patch.start()
```

Comment:
You need to stop the patch somewhere. For example add `self.addCleanup(self.base_dir_patch.stop)`

Reply:
This still applies.